File: src/main/java/com/keycloak/reactive/configuration/GlobalExceptionHandler.java
package com.keycloak.reactive.configuration;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestControllerAdvice;
import java.time.LocalDateTime;
import java.time.ZoneOffset;

@RestControllerAdvice
class GlobalExceptionHandler {

    @ExceptionHandler(Exception.class)
    ResponseEntity<ErrorResponse> handleException(Exception e) {
        return ResponseEntity
                .status(500)
                .body(
                        new ErrorResponse(
                                LocalDateTime.now().toEpochSecond(ZoneOffset.UTC),
                                500,
                                e.getMessage()
                        )
                );
    }

    record ErrorResponse(
        Long timestamp,
        int status,
        String message
    ) {}

}

File: src/main/kotlin/com/keycloak/reactive/configuration/GlobalExceptionHandler.kt
package com.keycloak.reactive.configuration

import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.RestControllerAdvice
import reactor.core.publisher.Mono
import java.time.LocalDateTime
import java.time.ZoneOffset

@RestControllerAdvice
class GlobalExceptionHandler {

    @ExceptionHandler(Exception::class)
    fun handleException(e: Exception) =
        ResponseEntity
            .status(500)
            .body(ErrorResponse(LocalDateTime.now().toEpochSecond(ZoneOffset.UTC), 500, e.message))

    data class ErrorResponse(
        val timestamp: Long,
        val status: Int,
        val message: String?
    )

}